components/core/SearchInput.tsx

```diff
@@ -19,26 +18,48 @@ const geocoder = new MapboxGeocoder({
 });
 
 function SearchInput(props: SearchInputProps): ReactElement {
-  const { placeholder, ariaLabel, onSubmit } = props;
+  const { placeholder, ariaLabel } = props;
   const router = useRouter();
+  const [searchInput, setSearchInput] = useState("");
+  const toast = useToast();
 
-  useEffect(() => {
-    const handleResult = (events: { result: Result }) => {
-      const { result } = events;
+  const handleResult = (events: { result: Result }) => {
+    const { result } = events;
 
-      router.push({
-        pathname: RoutePath.Search,
-        query: {
-          center: result.center,
-          place: result.place_name,
-          text: result.text,
-        },
-      });
-    };
+    toast.closeAll();
+
+    router.push({
+      pathname: RoutePath.Search,
+      query: {
+        center: result.center,
+        place: result.place_name,
+        text: result.text,
+      },
+    });
+  };
+
+  const handleError = () => {
+    toast({
+      id: "error",
+      title: "Location Not Found",
+      description: "Location entered couldn't be found, please try again.",
+      status: "error",
+      duration: 9000,
+      isClosable: true,
+    });
+  };
 
+  const onSubmit = () => {
+    geocoder.query(searchInput);
+    geocoder.on("result", handleResult);
+  };
+
+  useEffect(() => {
     geocoder.addTo("#location");
     geocoder.setPlaceholder(placeholder);
+    geocoder.on("loading", ({ query }) => setSearchInput(query));
     geocoder.on("result", handleResult);
+    geocoder.on("error", handleError);
 
     return () => {
       geocoder.off("result", handleResult);
```

Comment:
Generally it's a good idea to unbind/cleanup any listeners when a component is unmounted. So the return statement could look something like:

```javascript
return () => {
    geocoder.off("loading", handleLoad); // TODO: define/store load function in constant
    geocoder.off("result", handleResult);
    geocoder.off("error", handleError);
};
```

Reply:
fixed 63cc550